### PR TITLE
[node-group] Fix master node taint

### PR DESCRIPTION
**Pull request by yalosev**
Signed-off-by: Yuriy Losev <yuriy.losev@flant.com>

## Description
Fix node taints when NG master has `"node-role.kubernetes.io/master"` set explicitly

## Why do we need it, and what problem does it solve?
We have made fix for single node installations with taint removal but it can affect old cluster with only `"node-role.kubernetes.io/master"` taint set in the `master` NG

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-group
type: fix
summary: Avoid "node-role.kubernetes.io/master" taint removal when it is explicitly set in the master NG
```

### CODE REVIEW

**Comment by yalosev** [comment]
backport to 1.37.8 and 1.38.2

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/gcp'...

**Review by shvgn — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by name212** [comment]
@yalosev Approved but need to fix OpenApiTestCases

File: modules/040-node-manager/hooks/handle_node_templates.go [hunk 1]
```diff
@@ -225,20 +231,23 @@ func fixMasterTaints(sourceTaints []v1.Taint) []v1.Taint {
...
-	if _, ok := tmp["node-role.kubernetes.io/control-plane"]; !ok {
+	if _, ok := nodeTaintsMap["node-role.kubernetes.io/control-plane"]; !ok {
 		// control-plane taint was removed: single node installation
-		// also remove master taint if exists
-		if _, ok = tmp["node-role.kubernetes.io/master"]; ok {
-			delete(tmp, "node-role.kubernetes.io/master")
-			newTaints := make([]v1.Taint, 0, len(tmp))
-			for _, v := range tmp {
+		// also remove master taint if exists only in node spec.
+		// If master taint is set directly in the NG - keep it as is
```
Comment (shvgn) [inline modules/040-node-manager/hooks/handle_node_templates.go:237]: I would add an explanation why we do this.

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [2836](https://github.com/deckhouse/deckhouse/pull/2836) to the branch [release-1.37](https://github.com/deckhouse/deckhouse/tree/release-1.37) successful!

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [2837](https://github.com/deckhouse/deckhouse/pull/2837) to the branch [release-1.38](https://github.com/deckhouse/deckhouse/tree/release-1.38) successful!

**Pull request merged**